packages/graphql/src/schema/connection.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ConnectionArguments } from "graphql-relay";
+import { ArraySliceMetaInfo, getOffsetWithDefault, offsetToCursor } from "graphql-relay/connection/arrayConnection";
+
+/**
+ * Adapted from graphql-relay-js ConnectionFromArraySlice
+ */
+const createConnectionWithEdgeProperties = (
```

Comment:
So the `createConnectionWithEdgeProperties` function *is* an implementation of that algorithm. 

I'm not knowledgeable enough about cypher to know how to translate the algorithm for direct use. That said, with the exception of the `offsetToCursor` and `cursorToOffset` functions (see more discussion below), everything is pure math and logic checks. So it *should* be doable. Alas, despite lurking in the "#cypher" channel, I would need help with this

Reply:
So it looks like apoc already has base64 [encode](https://neo4j.com/labs/apoc/4.2/overview/apoc.text/apoc.text.base64Encode/) and [decode](https://neo4j.com/labs/apoc/4.2/overview/apoc.text/apoc.text.base64Decode/) procedures. 

So yes, I think it will be possible to write a cypher algorithm that does this. But this is probably something saved for one of your expertise.

Reply:
This was my best attempt at trying to write a cypher statement to create the cursors. I still haven't gotten it to work so no guarantee at all that it's correct. But maybe you or another neo4j all-star could use it as a starting point.

```
Match (this:MOVIE { title: "Point Break" })<-[this_acted_in:ACTED_IN]-(this_actor:Actor)
WITH this, this_acted_in, this_actor
WITH this, collect({ screenTime: this_acted_in.screenTime }) as edges, count(this_acted_in) as totalCount

CALL apoc.text.base64Decode("YXJyYXlDb25uZWN0aW9uNA==") YIELD prefixedCursor
CALL apoc.text.replace(prefixedCursor, "arrayConnection", "") YIELD stringValue
CALL apoc.number.parseInt(stringValue) YIELD intValue

CALL apoc.when(apoc.meta.isType(intValue, "INTEGER"), "RETURN intValue as offset", "RETURN 0 as offset") YIELD value

UNWIND edges as edge
UNWIND totalCount as i
CALL apoc.text.join(["arrayConnection", apoc.convert.toString(i + value.offset)]) YIELD cursorString
CALL apoc.text.base64Encode(cursorString) YIELD cursor
WITH collect(edge{.*, cursor: cursor }) as edges
RETURN { edges: edges, totalCount: size(edges) } AS actorsConnection
```

Reply:
Honestly, I probably wouldn't have done much better! 😝  There's definitely a lot of complexity there, and it would increase our reliance on APOC, which we're trying to avoid. Let's stick with the function implementation for the time-being, and keep this in our back pocket!